synapse/storage/databases/main/roommember.py

```diff
@@ -793,6 +793,21 @@ async def is_host_joined(self, room_id: str, host: str) -> bool:
     async def is_host_invited(self, room_id: str, host: str) -> bool:
         return await self._check_host_room_membership(room_id, host, Membership.INVITE)
 
+    async def is_locally_forgotten_room(self, room_id: str) -> bool:
+        sql = """
+            SELECT 1 FROM room_memberships
+            WHERE room_id = ?
+                AND user_id LIKE ?
```

Comment:
What are you doing with the `user_id`. Is it required?

There are other examples to get a bool from the database. E.g.:
https://github.com/matrix-org/synapse/blob/00a67f831adc7c01fcced6269cb5790816252493/synapse/storage/databases/main/registration.py#L443-L460

Reply:
My thinking with this is that remote rooms should be purged locally if all the local users have forgotten about it. I'm not 100% sure if purge works this way because I only tested it with local rooms anyways.

Reply:
In other words, it should not be required that remote users forget about a remote room, only local users are counted by this LIKE clause